### ISSUE TITLE
Regression: LineEdit loses focus after hitting enter.

### ISSUE THREAD

**Issue by jitspoe** [godotengine/godot#100230]
### Tested versions

- Reproducible in 4.4-dev6.
- Doesn't happen in 4.3 stable.

### System information

Godot v4.4.dev6 - Windows 10.0.19045 - Multi-window, 2 monitors - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3080 (NVIDIA; 32.0.15.6603) - AMD Ryzen 7 3700X 8-Core Processor (16 threads)

### Issue description

If you have a line edit and type something then press enter, you can no longer type in the edit box until you re-focus it by clicking or pressing enter again.

### Steps to reproduce

* Add a LineEdit to the scene.
* Type in the line edit.
* Press enter.
* Note that you can't type anymore.

(Very obvious with the dev console addon.  You can only type one command at a type).

### Minimal reproduction project (MRP)

[test_console.zip](https://github.com/user-attachments/files/18071805/test_console.zip)

**Comment by Sauermann** [comment]
Bisected to #87674

**Comment by akien-mga** [comment]
CC @WhalesState @KoBeWi

**Comment by WhalesState** [comment]
This feature was added to prevent the focus loss of the `LineEdit` when pressing `Escape`  button, a workaround for now is to use `LineEdit.edit().call_deferred()` when text is submitted, and also i can add a variable `keep_editing_on_text_submitted`.

**Comment by KoBeWi** [comment]
Yeah this is not really a regression, but a behavior change. The LineEdit is still usable.
I think adding `keep_editing_on_text_submitted` to easily preserve the old behavior makes sense.

**Comment by WhalesState** [comment]
> Yeah this is not really a regression, but a behavior change. The LineEdit is still usable. I think adding `keep_editing_on_text_submitted` to easily preserve the old behavior makes sense.

To avoid setting it to true for all the line edit nodes in editor, i have defaulted it to false for game (to avoid causing regression) and to true for editor and editor plugins.

**Comment by jitspoe** [comment]
> This feature was added to prevent the focus loss of the `LineEdit` when pressing `Escape` button, a workaround for now is to use `LineEdit.edit().call_deferred()` when text is submitted, and also i can add a variable `keep_editing_on_text_submitted`.

Not quite sure I understand -- the issue happens even if there's no callbacks for the lineedit.  You can just throw one in a scene and press enter and it loses focus.  Also, is the losing of focus on enter necessary to stop losing focus on escape?  Guess I don't fully understand the intent here.
